From 8410ab6de41a21eb0619f67ea36b4e2b8036dc5d Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Wed, 2 May 2018 13:33:37 +0300
Subject: [PATCH 3651/5242] MLK-18163-3: drm: imx: hdp: properly check DC bit
 depth

commit  d3e30ed3c73ca572b3d61304cc2b2024c46a482a from
https://source.codeaurora.org/external/imx/linux-imx.git

Currently, the code checks if RGB supports 10-bit when it should
actually check that YUV420 supports 10-bit.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
(cherry picked from commit 4064f377b0425fd7ec2c3ed15410ae7fad4077b5)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/hdp/imx-hdp.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/imx/hdp/imx-hdp.c b/drivers/gpu/drm/imx/hdp/imx-hdp.c
index cb66689..689927b 100644
--- a/drivers/gpu/drm/imx/hdp/imx-hdp.c
+++ b/drivers/gpu/drm/imx/hdp/imx-hdp.c
@@ -498,7 +498,7 @@ bool imx_hdp_bridge_mode_fixup(struct drm_bridge *bridge,
 
 	if (vic == VIC_MODE_97_60Hz &&
 	    (di->color_formats & DRM_COLOR_FORMAT_YCRCB420) &&
-	     di->bpc >= 10) {
+	    (di->hdmi.y420_dc_modes & DRM_EDID_YCBCR420_DC_30)) {
 		hdp->bpc = 10;
 		hdp->format = YCBCR_4_2_0;
 		return true;
-- 
1.7.9.5

